guidance/hybridpath_guidance/hybridpath_guidance/hybridpath_guidance_node.py

```diff
@@ -0,0 +1,140 @@
+import rclpy
+import numpy as np
+from rclpy.node import Node
+from nav_msgs.msg import Odometry
+from transforms3d.euler import quat2euler, euler2quat
+from geometry_msgs.msg import Pose2D
+from vortex_msgs.msg import HybridpathReference
+from hybridpath_guidance.hp_guidance_utils import odometrymsg_to_state, state_to_odometrymsg, Rot
+from hybridpath_guidance.hybridpath import HybridPathGenerator, HybridPathSignals
+
+class HybridPathGuidanceNode(Node):
+    def __init__(self, waypoints):
+        super().__init__("hybridpath_guidance_node")
+
+        self.declare_parameters(
+        namespace='',
+        parameters=[
+            ('hybridpath_guidance.lambda_val', 0.6),
+            ('hybridpath_guidance.path_generator_order', 2),
+            ('hybridpath_guidance.time_to_max_speed', 10.0),
+            ('hybridpath_guidance.dt', 0.05),
+        ])
+
+        self.state_subscriber_ = self.create_subscription(Odometry, "/sensor/seapath/odometry/ned", self.state_cb, 1)
+        self.guidance_publisher_ = self.create_publisher(HybridpathReference, "guidance/hybridpath/reference", 1)
+
+        # TODO: Add support for WaypointManager, add removal of "completed" waypoints
+
+        self.waypoints_ = waypoints
+        self.lambda_val_ = self.get_parameter('hybridpath_guidance.lambda_val').get_parameter_value().double_value
+        self.path_generator_order_ = self.get_parameter('hybridpath_guidance.path_generator_order').get_parameter_value().integer_value
+        self.time_to_max_speed_ = self.get_parameter('hybridpath_guidance.time_to_max_speed').get_parameter_value().double_value # Not used
+        self.dt_ = self.get_parameter('hybridpath_guidance.dt').get_parameter_value().double_value
+        
+        self.generator_ = HybridPathGenerator(waypoints, self.path_generator_order_, self.lambda_val_, 1)
+        self.path_ = self.generator_.Path
+
+        self.eta_d_ = np.zeros(3)
+        self.eta_d_prev_ = np.zeros(3)
+        self.eta_ = np.zeros(3)
+        self.eta_prev_ = np.zeros(3)
+        self.nu_ = np.zeros(3)
+        self.tau_ = np.zeros(3)
+        self.s_ = 0
+        self.u_desired_ = 1
+        
+        self.get_logger().info("hybridpath_guidance_node started")
+
+    def calculate_desired_states(self, dt: float) -> tuple[float, float, float, float, np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
This many return values with the same data types can get messy really quickly. I.e. it is very easy for a user to mix up the order of the outputs (yes, despite the docstring telling them what the outputs are........). If there is any way of packaging this in a separate data struct, I would highly recommend doing so